Newsletters: add 364 (2025-07-25) (#2425)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-07-25-newsletter.md

  Jesse Posner into improving the privacy and security of multisig
  collaborative custody.  In a typical collaborative custody service, a
  2-of-3 multisig may be used with the three keys being:

  - A user _hot key_ that is stored on a networked device and signs
    transactions for the user (either manually or through software
    automation).

  - A provider hot key that is stored on a separate networked device
    under the exclusive control of the provider.  The key only signs
    transactions according to a policy previously defined by the user,
    such as only allowing spending up to _x_ BTC a day.

  - A user _cold key_ that is stored offline and is only used if either
    the user hot key is lost or the provider ceases signing authorized
    transactions.

  Although the above configuration can provide a significant boost to
  security, the setup method almost always used involves the user
  sharing with the provider the [BIP32 extended public keys][topic
  bip32] for the user hot and cold wallets.  This allows the
  provider to detect all funds received to the user wallet and to track
  all spends of those funds even if the user spends without provider

>   assistance.  Several previous ways to mitigate this privacy loss have

Reviewer: ```suggestion
  assistance.  Several ways to mitigate this privacy loss have
```

>   been previously been described but they are either incompatible with

Reviewer: ```suggestion
 previously been described but they are either incompatible with
```

---
title: 'Bitcoin Optech Newsletter #364'
permalink: /en/newsletters/2025/07/25/
name: 2025-07-25-newsletter
slug: 2025-07-25-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME:harding

## News

- **LND gossip filter DoS vulnerability:** Matt Morehouse
  [posted][morehouse gosvuln] to Delving Bitcoin about a vulnerability
  affecting past versions of LND that he previously [responsibly
  disclosed][topic responsible disclosures].  An attacker could
  repeatedly request historic gossip messages from an LND node until it
  ran out of memory and was terminated.  The vulnerability was fixed in
  LND 0.18.3, released September 2024.

> - **Chain code witholding for multisig scripts:** Jurvis Tan

Reviewer 2: ```suggestion
- **Chain code withholding for multisig scripts:** Jurvis Tan
```

  [posted][tan ccw] to Delving Bitcoin about research he performed with
  Jesse Posner into improving the privacy and security of multisig
  collaborative custody.  In a typical collaborative custody service, a
  2-of-3 multisig may be used with the three keys being:

  - A user _hot key_ that is stored on a networked device and signs
    transactions for the user (either manually or through software
    automation).

  - A provider hot key that is stored on a separate networked device
    under the exclusive control of the provider.  The key only signs
    transactions according to a policy previously defined by the user,

>     such as only allowing spending up to _x_ BTC a day.

Reviewer 2: ```suggestion
    such as only allowing spending up to _x_ BTC per day.
```

>   - A user _cold key_ that is stored offline and is only used if either
>     the user hot key is lost or the provider ceases signing authorized

Reviewer 2: ```suggestion
  - A user's _cold key_ that is stored offline and is only used if either
    the hot key is lost, or the provider ceases signing authorized
```
Or "the user's hot key" (but I think it's clear it's the user's from earlier in the sentence).

Author: Leaving as-is.

    transactions.

  Although the above configuration can provide a significant boost to
  security, the setup method almost always used involves the user
  sharing with the provider the [BIP32 extended public keys][topic

>   bip32] for the user hot and cold wallets.  This allows the

Reviewer 2: ```suggestion
  bip32] for the user's hot and cold wallets.  This allows the
```

>   provider to detect all funds received to the user wallet and to track

Reviewer 2: ```suggestion
  provider to detect all funds received into the user's wallet and to track
```
Or maybe "received by the user's wallet"

>   all spends of those funds even if the user spends without provider

Reviewer 2: ```suggestion
  all spends of those funds, even if the user spends without the provider's
```

  assistance.  Several previous ways to mitigate this privacy loss have

>   been previously been described but they are either incompatible with

Reviewer 2: ```suggestion
  been described, but they are either incompatible with
```

  typical use (e.g. using separate tapleaves) or complex (e.g. requiring
  [MPC][]).  Tan and Posner describe a simple alternative:

  - The provider generates half of a BIP32 HD extended key (just
    the key part).  They give the public key to the user.

  - The user generates the other half (the chain code).  They keep this
    private.

  When receiving funds, the user can combine the two halves to create an

>   extended public key (xpub) and then derive multisig addresses like
>   normally.  The provider doesn't know the chain code, preventing them

Reviewer 2: ```suggestion
  extended public key (xpub) and then derive multisig addresses as
  usual.  The provider doesn't know the chain code, preventing them
```

  from deriving the xpub or discovering the address.

  When spending funds, the user can derive from the chain code the
  necessary _tweak_ that the provider needs to combine with their

>   private key in order to create a valid signature.  They simply share

Reviewer 2: ```suggestion
  private key to create a valid signature.  They simply share
```

  this tweak with the provider.  The provider can't learning anything

>   from the tweak accept that it's valid for spending from the specific

Reviewer 2: ```suggestion
  from the tweak except that it's valid for spending from the specific
```

>   scriptPubKey the received funds.

Reviewer 2: ```suggestion
  scriptPubKey that received funds.
```

>   Some providers may require that the change output for a spending
>   transaction sends the money back to the same script template.  Tan's

Reviewer 2: ```suggestion
  Some providers may require that the spending transaction's change output
  sends the money back to the same script template.  Tan's
```

>   post describes how this can easily be accomplished.

Reviewer 2: ```suggestion
  post describes how his proposal easily accomplishes this.
```

> - **Research indicates common Bitcoin primitives compatible with quantum resistant signature algorithms:**

Reviewer 2: ```suggestion
- **Research indicates common Bitcoin primitives are compatible with quantum-resistant signature algorithms:**
```

>   Jesse Posner [posted][posner qc] to Bitcoin-Dev several links to

Reviewer 3: ```suggestion
  Jesse Posner [posted][posner qc] to Delving Bitcoin several links to
```

Author: D'oh.  Thanks for catching that!

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [How does Bitcoin Core handle reorgs larger than 10 blocks?]({{bse}}127512)
  TheCharlatan links to Bitcoin Core code that handles chain reorganizations by
  only re-adding a maximum of 10 blocks worth of transactions to the mempool.

- [Advantages of a signing device over an encrypted drive?]({{bse}}127596)
  RedGrittyBrick points out that data on an encrypted drive can be extracted
  while the drive is unencrypted whereas hardware signing devices are designed to
  prevent this data extraction attack.

- [Spending a taproot output through the keypath and scriptpath?]({{bse}}127601)
  Antoine Poinsot details how the use of merkle trees, key tweaks, and leaf

>   scripts achieve taproot's keypath and scriptpath spending capabilities.

Reviewer 2: ```suggestion
  scripts achieves taproot's keypath and scriptpath spending capabilities.
```
I think, because "the use of" is singular.

Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #32521][] makes legacy transactions with more than 2500
  signature operations (sigops) non-standard in preparation for a potential
  [consensus cleanup soft fork][topic consensus cleanup] upgrade that would
  enforce the limit at the consensus level. If the softfork took place without
  this change, miners that don’t upgrade could become targets of trivial DoS
  attacks. See Newsletter [#340][news340 sigops] for additional details on the
  legacy input sigops limit.

- [Bitcoin Core #31829][] adds resource limits to the orphan transaction
  handler, `TxOrphanage` (See Newsletter [#304][news304 orphan]), to preserve
  opportunistic one-parent-one-child (1p1c) [package relay][topic package relay]
  in the face of DoS spam attacks. Four limits are enforced: a global cap
  of 3 000 orphan announcements (CPU and latency cost), a proportional per‑peer

>   orphan announcements cap, a per‑peer weight reservation of 24 × 400 kWU and a

Reviewer 2: ```suggestion
  orphan announcements cap, a per‑peer weight reservation of 24 × 400 kWU, and a
```

  oldest orphan announcement from the peer that has used the most CPU or memory
  relative to its allowance (highest Peer DoS Score). The PR also deletes the
  `‑maxorphantxs` option (default 100), whose policy of evicting random
  announcements allowed attackers to replace the entire orphan set and render
  [1p1c relay][1p1c relay] useless.

- [LDK #3801][] extends [attributable failures][topic attributable failures] to
  the payment success path by recording how long a node holds an [HTLC][topic
  htlc] and propagating those hold‑time values upstream in the attribution
  payload. Previously, LDK only tracked hold times for failed payments (see
  Newsletter [#349][news349 attributable]).

- [LDK #3842][] extends its [interactive transaction construction][topic dual
  funding] state machine (See Newsletter [#295][news295 dual]) to handle the
  signing coordination for shared inputs in [splicing][topic splicing]
  transactions. The `prevtx` field of the `TxAddInput` message is made optional
  to reduce memory usage and simplify validation.

- [BIPs #1890][] changes the separator parameter from `+` to `-` in [BIP77][]

>   because some HTML 2.0 URI libraries treat `+` as if it was a blank space. In

Reviewer 2: ```suggestion
  because some HTML 2.0 URI libraries treat `+` as if it were a blank space. In
```
